### PR TITLE
session: refine non-transactional delete

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

Issue Number: ref #33485

Problem Summary:

### What is changed and how it works?

Refine the feature in:
- telemetry
- logging
- constraint check: limit/order by clause, prepare protocol
- memory tracking
- table alias

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/nontransactional.go [hunk 1]
```diff
@@ -336,7 +360,7 @@ func buildShardJobs(ctx context.Context, stmt *ast.NonTransactionalDeleteStmt, s
 		if chk.NumRows() == 0 {
 			if currentSize > 0 {
 				// there's remaining work
-				jobs = append(jobs, job{jobID: jobCount, start: currentStart, end: currentEnd, jobSize: currentSize})
+				jobs = appendNewJob(jobs, jobCount+1, currentStart, currentEnd, currentSize, memTracker)
```
Comment (cfzjywxk) [inline session/nontransactional.go:363]: Why another plus one here?

Comment (ekexium) [reply session/nontransactional.go:363]: To make the job id 1..N, instead of 0..N-1

**Review by sticnarf — APPROVED** [review]
LGTM. Please resolve the conflicts.

**Pull request merged**